src/app.js

```diff
@@ -42,38 +42,31 @@ app.get('/products', (req, res) => {
     })
 })
 
-app.get('/weather', (req, res) => {
+app.get('/weather', async (req, res) => {
```

Comment:
I'm fairly new to async/await so I am not totally clear on how to use it, is there any benefit of using this here than what I originally write? for instance, it can be faster?

Reply:
There is no benefit in speed. Both what you had and this will work equally as fast. The main benefit is having easier to read code.
Nesting many statements together with callbacks leads to callback hell really quickly, and async await and promises have made callback hell much easier to avoid.
The other thing that this code does better is it puts all the error handling code in one place so you no longer need to duplicate that in every callback. It also makes handling errors easier since you don't need to have the callback return undefined for either the error or data portion of the callback.

I generally prefer to use async/await and promises over callbacks when possible due to the usually cleaner nature of them.